rope/base/pyobjects.py

```diff
@@ -22,7 +23,7 @@ def get_attribute(self, name):
     def get_module(self):
         return None
 
-    def get_type(self):
+    def get_type(self) -> PyObject:
```

Comment:
This is probably going to require a bit of re-thinking and re-architecturing here with `get_type()`. `get_type()` is supposed to return the inferred type of an object, so the intended case here is that it should return some instance of AbstractClass (e.g. int = BuiltinClass, or user-defined class = PyClass). 

However, in the current implementation, when rope cannot infer the type of an object, the current implementation would just return just some empty PyObject as a placeholder, which doesn't look correct/ideal to me. Maybe we could have some sort of UnknownClass instead to more explicitly describe what's happening; and then this method then can instead be annotated as such:

```
def get_type(self) -> AbstractClass:
    ...
```